### PR TITLE
feat(training): add SurgeFakeOracleModule for oracle-baseline smoke tests

### LINKED ISSUE #1068: feat(training): add SurgeFakeOracleModule for oracle-baseline smoke tests
Add a `SurgeFakeOracleModule` Lightning module that is a drop-in replacement for `SurgeFeedForwardModule` and returns `batch["params"]` as predictions (perfect-inversion oracle).

Switch the `_build_surge_xt_smoke_cfg` fixture in `tests/conftest.py` from `experiment=surge/ffn_full` → `experiment=surge/fake_oracle` so the smoke tests `test_train_surge_xt` and `test_train_eval_surge_xt` exercise the oracle instead of an untrained real model.

Motivation:
- Smoke tests today wire up a real (but untrained) Surge FFN that randomly initializes a 100M-param transformer just to be discarded after one step — wastes setup time and is sensitive to any deep regression in the production net.
- The oracle isolates the smoke tests from production-model changes, lets us validate the predict→render→audio-metrics pipeline against a known-correct upstream, and establishes a clean "what's the best downstream could ever look like" baseline.

Sub-issue of Phase 4: Test Coverage (#30).

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary

- Add `SurgeFakeOracleModule` — a drop-in replacement for `SurgeFeedForwardModule` whose `predict_step` returns `batch["params"]` verbatim (perfect-inversion oracle). Backing `FakeOracleNet` holds one dummy `nn.Parameter` so the optimizer / `trainer.fit()` keep working.
- Add Hydra wiring: `configs/model/surge_fake_oracle.yaml` + `configs/experiment/surge/fake_oracle.yaml` mirroring the `surge_ffn` / `surge/ffn_full` shape so the cfg path `model.net.d_out` / `model.compile` / `model.scheduler` keep resolving for the conftest bake-ins.
- Switch `_build_surge_xt_smoke_cfg` in `tests/conftest.py` from `experiment=surge/ffn_full` → `experiment=surge/fake_oracle`. Update `configs/experiment/surge/test-mps.yaml` in lockstep so `test_test_mps_yaml_matches_cfg_surge_xt_global` (the drift guard) stays green.
- 17 new unit tests in `tests/models/test_surge_fake_oracle_module.py` covering the predict-step contract, mel-spec independence, model_step shapes, zero-loss-with-grad, val/test dict shape (consumed by `LogPerParamMSE`), and B=1/B=4 invariance.

## Test plan

- [x] `make test-fast` — 799 passed, 2 skipped (matches baseline).
- [x] `pytest tests/models/test_surge_fake_oracle_module.py -v` — 17/17 passed.
- [x] `pytest tests/test_train.py::test_cfg_surge_xt_global_wires_param_spec tests/test_configs.py::test_test_mps_yaml_matches_cfg_surge_xt_global -v` — 4/4 passed.
- [x] `pytest tests/test_train.py::test_train_surge_xt tests/test_train.py::test_train_eval_surge_xt -v` — both pass on cpu and gpu locally (MPS errors are the standard "no MPS hardware" hardfail).
- [ ] CI green on this PR.

Closes #1068.

### CODE REVIEW

**Comment by ktinubu** [comment]
## Verification Results

PR promise: a `SurgeFakeOracleModule` that is a drop-in for `SurgeFeedForwardModule`, with `predict_step` returning `batch["params"]` verbatim, swapped into the smoke fixture so `test_train_surge_xt` / `test_train_eval_surge_xt` exercise the oracle.

All evidence below was captured locally in worktree `feat/surge-fake-oracle-module` at HEAD `460cee1` (`/usr/lib/vst3/Surge XT.vst3` symlinked into the worktree's `plugins/` for VST tests). CI on the same SHA is in flight on the PR.

- [x] **`test_train_eval_surge_xt[cpu]` runs the full predict → VST-render → audio-metrics pipeline against the oracle and produces finite metrics** *(Level 1 — promise is end-to-end smoke coverage of the predict-side pipeline; lower levels would not exercise PredictionWriter, predict_vst_audio, or compute_audio_metrics)*
  ```
  $ pytest tests/test_train.py::test_train_eval_surge_xt -v --no-header -k "cpu"
  tests/test_train.py::test_train_eval_surge_xt[cpu] PASSED
  ============= 1 passed, 5 deselected ============= (call: 48.07s)
  ```
- [x] **`test_train_eval_surge_xt[gpu]` same pipeline on CUDA** *(Level 1 — repeat on GPU to catch device-placement regressions in the oracle's `batch["params"]` passthrough)*
  ```
  $ pytest tests/test_train.py::test_train_eval_surge_xt -v --no-header -k "gpu"
  tests/test_train.py::test_train_eval_surge_xt[gpu] PASSED
  GPU probe: NVIDIA GeForce RTX 4090, 24546 MiB free; torch.cuda.is_available()=True
  ```
- [x] **`test_train_surge_xt[cpu, gpu]` — `trainer.fit()` advances and `train/loss*` is finite on the oracle** *(Level 1 — exercises the full Lightning fit loop, configure_optimizers, dummy-param grad path, and train-loss logging end-to-end)*
  ```
  $ pytest tests/test_train.py::test_train_surge_xt -v --no-header -k "cpu or gpu"
  tests/test_train.py::test_train_surge_xt[cpu] PASSED
  tests/test_train.py::test_train_surge_xt[gpu] PASSED
  ```
- [x] **`predict_step` oracle contract: returns `(batch["params"], batch)` and ignores `mel_spec`** *(Level 2 — invokes the method directly with controlled inputs; covers the property `LogPerParamMSE` and `PredictionWriter` depend on)*
  ```
  $ pytest tests/models/test_surge_fake_oracle_module.py -v --no-header
  17 passed in 0.47s
  # Includes:
  # test_predict_step_returns_params_and_batch_tuple[1,4]
  # test_predict_step_ignores_mel_spec[1,4]
  # test_model_step_returns_four_tuple_with_oracle_preds[1,4]
  # test_model_step_loss_is_exactly_zero[1,4]
  # test_model_step_loss_requires_grad[1,4]
  # test_training_step_returns_zero_loss_with_grad[1,4]  (asserts net.dummy.grad lands after backward)
  # test_validation_step_returns_zero_per_param_mse[1,4]
  # test_test_step_returns_zero_per_param_mse[1,4]
  # test_fake_oracle_net_exposes_d_out_and_has_grad_parameter
  ```
- [x] **`test-mps.yaml` drift guard stays green after the in-lockstep update** *(Level 2 — runs the live composition equality test; catches accidental skew between the conftest bake-ins and the YAML peers)*
  ```
  $ pytest tests/test_configs.py::test_test_mps_yaml_matches_cfg_surge_xt_global -v --no-header
  PASSED in 0.15s
  ```
- [x] **`model.net.d_out` and `callbacks.log_per_param_mse.param_spec` continue to propagate through the smoke fixture for all 3 supported param specs** *(Level 2 — runs the parametrized contract test that pins the fixture's parametrization plumbing)*
  ```
  $ pytest tests/test_train.py::test_cfg_surge_xt_global_wires_param_spec -v --no-header
  test_cfg_surge_xt_global_wires_param_spec[surge_4] PASSED
  test_cfg_surge_xt_global_wires_param_spec[surge_simple] PASSED
  test_cfg_surge_xt_global_wires_param_spec[surge_xt] PASSED
  ```
- [x] **Pre-commit + format clean on the committed state** *(Level 2 — runs the hooks; not a standalone "make format" checkbox, this is the gate that proves the typing / annotation rewrites survive ruff + pydoclint + pyright in this PR's diff)*
  ```
  $ make format 2>&1 | tail -8
  pyright..................................................................Passed
  docformatter.............................................................Passed
  interrogate..............................................................Passed
  pydoclint................................................................Passed
  prettier.................................................................Passed
  mdformat.................................................................Passed
  codespell................................................................Passed
  ```
- [ ] **`test_train_surge_xt[mps]` / `test_train_eval_surge_xt[mps]`** *(SKIP — host has no MPS hardware; covered by the macOS runner in CI's `run_tests_macos` lane.)*
  To verify manually on an Apple Silicon machine:
  ```
  pytest tests/test_train.py::test_train_surge_xt tests/test_train.py::test_train_eval_surge_xt -k mps
  ```
  GPU probes for the SKIP rationale (per CLAUDE.md "GPU Verification"):
  ```
  $ python3 -c "import torch; print('mps avail:', torch.backends.mps.is_available(), 'built:', torch.backends.mps.is_built())"
  mps avail: False built: False
  ```

File: pyrightconfig.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "src/synth_setter/models/ksin_ff_module.py",
     "src/synth_setter/models/ksin_flow_matching_module.py",
     "src/synth_setter/models/surge_ff_module.py",
+    "src/synth_setter/models/surge_fake_oracle_module.py",
     "src/synth_setter/models/surge_flow_matching_module.py",
     "src/synth_setter/models/surge_flowvae_module.py",
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply pyrightconfig.json:32]: Fixed in c93b4e5 — switched `self.hparams.foo` to `self.hparams["foo"]` indexing and replaced `self.trainer.model.parameters()` with `self.parameters()`. Pyright now passes cleanly on the file with no excludes.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -70,6 +70,7 @@ repos:
             src/synth_setter/models/ksin_ff_module\.py|
             src/synth_setter/models/ksin_flow_matching_module\.py|
             src/synth_setter/models/surge_ff_module\.py|
+            src/synth_setter/models/surge_fake_oracle_module\.py|
             src/synth_setter/models/surge_flow_matching_module\.py|
             src/synth_setter/models/surge_flowvae_module\.py|
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply .pre-commit-config.yaml:74]: Fixed in c93b4e5 — same pyright cleanup as the pyrightconfig.json comment; surge_fake_oracle_module is no longer in this regex.

File: src/synth_setter/models/surge_fake_oracle_module.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    """LightningModule whose predictions are an oracle copy of ``batch["params"]``."""
+
+    def __init__(
+        self,
+        net: nn.Module,
+        optimizer: torch.optim.Optimizer,
+        scheduler: torch.optim.lr_scheduler,
+        compile: bool = False,
+        warmup_steps: int = 0,
+    ):
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply src/synth_setter/models/surge_fake_oracle_module.py:60]: Fixed in 460cee1 — annotations are now `Callable[..., torch.optim.Optimizer]` and `Callable[..., torch.optim.lr_scheduler.LRScheduler] | None` with a `None` default. The test-side `# type: ignore[arg-type]` was removed at the same time, and c93b4e5 dropped the pyright exclude.

File: tests/conftest.py [hunk 4]
```diff
@@ -305,7 +305,7 @@ def _build_surge_xt_smoke_cfg(accelerator: str, param_spec_name: str) -> DictCon
         cfg = compose(
             config_name="train.yaml",
             return_hydra_config=True,
-            overrides=["experiment=surge/ffn_full", "callbacks=[default_surge,eval_surge]"],
+            overrides=["experiment=surge/fake_oracle", "callbacks=[default_surge,eval_surge]"],
         )
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply tests/conftest.py:309]: Fixed in 731c3aa — docstring now says `experiment=surge/fake_oracle`. The PR also includes a sibling commit updating two stale references in docs/reference/testing.md.

File: tests/models/test_surge_fake_oracle_module.py [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+def _make_module() -> SurgeFakeOracleModule:
+    """Build a fresh :class:`SurgeFakeOracleModule` with a partial Adam optimizer.
+
+    :return: Module ready for direct step-method calls (no Trainer attached).
+    :rtype: SurgeFakeOracleModule
+    """
+    net = FakeOracleNet(d_out=_NUM_PARAMS)
+    optimizer = partial(torch.optim.Adam, lr=1e-4)
+    return SurgeFakeOracleModule(net=net, optimizer=optimizer, scheduler=None)  # type: ignore[arg-type]
+
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply tests/models/test_surge_fake_oracle_module.py:68]: Fixed in 460cee1 — the upstream `__init__` annotations were tightened to `Callable[..., Optimizer]` and `Callable[..., LRScheduler] | None`, and this `# type: ignore` was removed in the same commit. c93b4e5 then dropped the pyright exclude for the module.

File: pyrightconfig.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ktinubu) [reply pyrightconfig.json:32]: Fixed in c93b4e5 — surge_fake_oracle_module is no longer in pyright's exclude list, and pyright now type-checks the file cleanly (after switching to dict-style `self.hparams[...]` access and `self.parameters()`).

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ktinubu) [reply .pre-commit-config.yaml:74]: Fixed in c93b4e5 — surge_fake_oracle_module removed from this regex as well; the module is fully type-checked by both the pyrightconfig.json and pre-commit pyright hooks.

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -151,6 +152,7 @@ repos:
...
             src/synth_setter/utils/callbacks\.py
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply .pre-commit-config.yaml:156]: Fixed in c93b4e5 — surge_fake_oracle_module was also dropped from the interrogate exclude block in the same commit. The module's public surface has full docstrings (interrogate-clean).
